OpenRA.Mods.Common/Traits/Render/WithDeathAnimation.cs

```diff
@@ -61,20 +64,34 @@ public static object LoadDeathTypes(MiniYaml yaml)
 	public class WithDeathAnimation : INotifyKilled
 	{
 		public readonly WithDeathAnimationInfo Info;
+		readonly Crushable crushable;
 		readonly RenderSprites rs;
 
 		public WithDeathAnimation(Actor self, WithDeathAnimationInfo info)
 		{
 			Info = info;
 			rs = self.Trait<RenderSprites>();
+			crushable = self.TraitOrDefault<Crushable>();
```

Comment:
You should do this lookup in the `Killed` method, since there is no guarantee we will initialize the crushable trait before this one.